### ISSUE TITLE
[List/List item] Context not provided to `selected` list items if previously selected

### ISSUE THREAD

**Issue by geospatialem** [Esri/calcite-components#6711]
### Summary

Context should be provided to `list-items` selections if a selection is added on the component's initialization, or a user returns to the component later to get context on the selection(s).

### Actual Behavior

Context is provided if the user adds selection to a `list-item`, however **no context is provided to previously selected or `list-items` selected on initialization**. 

cc: @FelFly 

### Expected Behavior

`list-item`'s with selection should be announced if the `selected` attribute/prop is present, and as a user interacts with the `list` and updates the selection.

### Reproduction Sample

https://codepen.io/geospatialem/pen/wvELvgr

### Reproduction Steps

1. Open the [sample](https://codepen.io/geospatialem/pen/wvELvgr)
2. Activate a screen reader of choice (VO with Safari, JAWS with Chrome/FF, NVDA with Firefox)
3. Observe the screen reader will not announce if a `list-item` is already `selected`.
4. Select a new `list-item`
5. Observe the selection is announced.

### Reproduction Version

1.2.0

### Relevant Info

Applies to JAWS and NVDA on Windows and VoiceOver on Mac.

### Priority impact

p2 - want for current milestone

### Esri team

Calcite (dev)

**Comment by geospatialem** [comment]
Research will be conducted in the April milestone to determine the feasibility for a fix.
